deckhand/control/documents.py

```diff
@@ -0,0 +1,79 @@
+# Copyright 2017 AT&T Intellectual Property.  All other rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import yaml
+
+import falcon
+
+from oslo_log import log as logging
+
+from deckhand.control import base as api_base
+from deckhand.engine import document_validation
+from deckhand import errors as deckhand_errors
+from deckhand.objects import documents
+
+LOG = logging.getLogger(__name__)
+
+
+class DocumentsResource(api_base.BaseResource):
+    """API resource for realizing CRUD endpoints for Documents."""
+
+    def __init__(self, **kwargs):
+        super(DocumentsResource, self).__init__(**kwargs)
+        self.authorized_roles = ['user']
+
+    def on_get(self, req, resp):
+        pass
+
+    def on_head(self, req, resp):
+        pass
+
+    def on_post(self, req, resp):
+        """Create a document. Accepts YAML data only."""
+        if req.content_type != 'application/yaml':
```

Comment:
Nice catch, I noticed you added that in your design document, I just haven't gotten around to changing it. I'll use a global variable so that way this sort of typo won't be made in the future.